### DISCUSSION TITLE
WASM

### DISCUSSION THREAD

**Discussion by RKlunares** [unoplatform/uno#5342] [Q&A]
Hey Uno Team,

I have carefully studied your examples for WASM, JavaScript interaction and embedding JS sources. I have also studied your sources here. You really provide a good document system. However, there is one thing that remains a mystery to me despite your documentation efforts. 
If I want to include JS sources, I naturally want to access the JS source and also want to pick up the values. Furthermore, I want to exchange array contents directly and make them available to the JS source and of course get them back. With you I can execute JS but I only get a string back, and no JSObject as expected, which can be used to call other methods. In the webassembly world, Mono, this is not a problem. However, when I include Webassembly, i.e. the Mono sources, I get a compile error. I see from your sources that you probably use something similar. Unfortunately, your sources are marked as internal. I can therefore not access them. I urgently need access to the web assembly area, or at least the same access rights. 

I hope you can help me or you have a solution :) 

Thank you very much for your help. 

RK

Translated with www.DeepL.com/Translator (free version)

**Comment by jeromelaban** [comment]
Thanks for reaching out! 

Could you try to create a simple repro project (using a uno "cross-platform app" template)  with what you're trying to achieve ? You're unfortunately not giving enough information for us to help you.

**Reply by RKlunares** [reply]
Hi, 

I think creating a project for this would be a bit much. 
It is actually quite simple. I want to include the Nuget Packet Webassembly.Bindings (https://www.nuget.org/packages/WebAssembly.Bindings/) in the Uno Wasm project to use something like JSObject and Runtime. 
However, when I compile, I get the error that the Webassembly.Bindings assembly already exists in the OBJ directory.

Org. Exception: 

1>.nuget\packages\uno.wasm.bootstrap\1.3.4\build\Uno.Wasm.Bootstrap.targets(126,2): error : [00:00:03.9288819]
1>.nuget\packages\uno.Wasm.Bootstrap\1.3.4\build\Uno.Wasm.Bootstrap.targets(126,2): error : [00:00:03.9313472] 

Unhandled exception: System.IO.IOException: the file "\.....\obj\Debug\netstandard2.0\dist_work\managed\WebAssembly.Bindings.dll" already exists.

1>.nuget\packages\uno.wasm.bootstrap\1.3.4\build\Uno.Wasm.Bootstrap.targets(126,2): error : [00:00:03.9313726] at System.IO.__Error.WinIOError(Int32 errorCode, String maybeFullPath)
1>.nuget\packages\uno.Wasm.Bootstrap\1.3.4\build\Uno.Wasm.Bootstrap.targets(126,2): error : [00:00:03.9313764] on System.IO.File.InternalCopy(String sourceFileName, String destFileName, Boolean overwrite, Boolean checkHost)
1>.nuget\packages\uno.Wasm.Bootstrap\1.3.4\build\Uno.Wasm.Bootstrap.targets(126,2): error : [00:00:03.9313781] on Driver.CopyFile(String sourceFileName, String destFileName, CopyType copyType, String typeFile)
1>.nuget\packages\uno.Wasm.Bootstrap\1.3.4\build\uno.Wasm.Bootstrap.targets(126,2): error : [00:00:03.9313810] on Driver.Run(String[] args)
1>.nuget\packages\uno.Wasm.Bootstrap\1.3.4\build\Uno.Wasm.Bootstrap.targets(126,2): error : [00:00:03.9314060] on Driver.Main(String[] args)
1>.nuget\packages\uno.Wasm.Bootstrap\1.3.4\build\Uno.Wasm.Bootstrap.targets(126,2): error : System.Exception: Failed to generate wasm layout (More details are available in diagnostic mode or with the MSBuild switch /bl)
1>.nuget\packages\uno.wasm.bootstrap\1.3.4\build\Uno.Wasm.Bootstrap.targets(126,2): error : at Uno.Wasm.Bootstrap. ShellTask_v2ed6794b633a8ebf33d51d6d21f9d6a587ebe09e.RunPackager() in D:\a\1\s\src\Uno.Wasm.Bootstrap\ShellTask.cs:line 583.
1>.nuget\packages\uno.Wasm.Bootstrap\1.3.4\build\Uno.Wasm.Bootstrap.targets(126,2): error : at Uno.Wasm.Bootstrap. ShellTask_v2ed6794b633a8ebf33d51d6d21f9d6a587ebe09e.Execute() in D:\a\1\s\src\Uno.Wasm.Bootstrap\ShellTask.cs:Zeile 182.

The beginnings of the file paths are anonymised

It would be a great advantage if I could also use the Webassembly.Bindings. What is the best way to do this and is this a bug in the build task or why is the possibility so blocked?

Many thanks for your efforts :) 

Kind regards

Rolf

**Reply by jeromelaban** [reply]
Thanks for the report. The WebAssembly.Bindings package is not supported at this point, as are all the associated APIs provided there. We'll be providing support in the future once .NET 6 is giving access to it in a more developer friendly way.
